doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,400 @@
+    This guide is relevant for all trainers in which a custom training loop is defined. 
+    This includes :class:`TorchTrainer <ray.train.torch.TorchTrainer>` and 
+    :class:`TensorflowTrainer <ray.train.tensorflow.TensorflowTrainer>`.
 
-Ray Train also exposes logging callbacks that automate some of these tasks.
+Most experiment tracking libraries work out-of-the-box with Ray Train. 
+This guide provides instructions on how to set up the code so that your favorite experiment tracking libraries 
+can work for distributed training with Ray Train. The end of the guide has common errors to aid in debugging 
+the setup.
 
-.. _train-experiment-tracking-native:
+The following psudo code demonstrates how to use the native experiment tracking library calls 
+inside of Ray Train: 
 
-Using native experiment tracking libraries
-------------------------------------------
+.. code-block:: python
 
-.. tab-set::
+    from ray.train.torch import TorchTrainer
+    from ray.train import ScalingConfig
 
-    .. tab-item:: PyTorch
+    def train_func(config):
+        # Training code and native experiment tracking library calls go here.
 
-        You can use experiment tracking libraries such as Weights & Biases, Mlflow, or
-        Comet directly in your Ray Train training loop.
+    scaling_config = ScalingConfig(num_workers=2, use_gpu=True)
+    trainer = TorchTrainer(train_func, scaling_config=scaling_config)
+    result = trainer.fit()
 
-        There are two things to keep in mind:
+Ray Train lets you use native experiment tracking libraries by customizing the tracking 
+logic inside the ``train_func`` function. In this way, you can port your experiment tracking 
+logic to Ray Train with minimal changes. 
 
-        1. Your code is executed in parallel on many workers. However, you often only want to report
-        results from one of these workers (usually the first worker - the "rank 0" worker).
+Getting Started
+===============
 
-        2. When using the native libraries, you should report the results to both Ray Train and
-        the experiment tracking library.
+Let's start by looking at some code snippets.
 
-        Example:
+The following session uses Weights & Biases (W&B) and MLflow but it is adaptable to other frameworks.
 
-        .. code-block:: python
+.. tabs::
+
+    .. tab:: W&B
 
+        .. code-block:: python
+            
+            import ray
             from ray import train
+            import wandb
+
+            # This ensures that all ray worker processes have `WANDB_API_KEY` set.
+            ray.init(runtime_env={"env_vars": {"WANDB_API_KEY": "your_api_key"}})
 
-            def train_fn(config):
-                context = train.get_context()
+            def train_func(config):
+                # Step 1 and 2
+                if train.get_context().get_world_rank() == 0:
+                    wandb.init(
+                        name=...,
+                        project=...,
+                        # ...
```

Comment:
Use `id` here since it's what you refer to in step 2.

Reply:
I don't think it's a big deal. name and project etc are just more intuitive when referencing the run. For that reason, that's what you see if you open wandb website and click only one example. And we should try to keep that intuitiveness. 
id is what you do when you care about logically continue the last run (a tip about fault tolerant, a more advanced case).